### ISSUE TITLE
build error: Creation method of abstract class cannot be public 

### ISSUE THREAD

**Issue by anoduck** [steveno/libbalistica#10]
Attempted to build and received an odd error that I have never seen before. 

```
[13/35] Compiling Vala source ../test/ConversionTests.vala ../test/LibBalisti...on/Mass.vala ../lib/conversion/Angle.vala ../lib/conversion/Temperature.vala.
FAILED: test/9f86d08@@libbalistica-test@exe/ConversionTests.c test/9f86d08@@libbalistica-test@exe/LibBalisticaTests.c test/9f86d08@@libbalistica-test@exe/MillerTests.c test/9f86d08@@libbalistica-test@exe/GreenhillTests.c test/9f86d08@@libbalistica-test@exe/TestRunner.c test/9f86d08@@libbalistica-test@exe/AbstractTestCase.c test/9f86d08@@libbalistica-test@exe/Atmosphere.c test/9f86d08@@libbalistica-test@exe/Greenhill.c test/9f86d08@@libbalistica-test@exe/Miller.c test/9f86d08@@libbalistica-test@exe/Retard.c test/9f86d08@@libbalistica-test@exe/Solve.c test/9f86d08@@libbalistica-test@exe/Zero.c test/9f86d08@@libbalistica-test@exe/LibBalistica.c test/9f86d08@@libbalistica-test@exe/Pbr.c test/9f86d08@@libbalistica-test@exe/Solution.c test/9f86d08@@libbalistica-test@exe/Windage.c test/9f86d08@@libbalistica-test@exe/Mass.c test/9f86d08@@libbalistica-test@exe/Angle.c test/9f86d08@@libbalistica-test@exe/Temperature.c 
valac -C --debug --debug --pkg gee-0.8 --target-glib 2.40.0 --pkg glib-2.0 --color=always --directory test/9f86d08@@libbalistica-test@exe --basedir ../test ../test/ConversionTests.vala ../test/LibBalisticaTests.vala ../test/MillerTests.vala ../test/GreenhillTests.vala ../test/TestRunner.vala ../test/AbstractTestCase.vala ../lib/Atmosphere.vala ../lib/Greenhill.vala ../lib/Miller.vala ../lib/Retard.vala ../lib/Solve.vala ../lib/Zero.vala ../lib/LibBalistica.vala ../lib/Pbr.vala ../lib/Solution.vala ../lib/Windage.vala ../lib/conversion/Mass.vala ../lib/conversion/Angle.vala ../lib/conversion/Temperature.vala
../test/AbstractTestCase.vala:11.4-11.26: error: Creation method of abstract class cannot be public.
   public AbstractTestCase (string name) {
   ^^^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 1 error(s), 0 warning(s)
[14/35] Compiling C object 'lib/76b5a35@@balistica@sha/meson-generated_Solution.c.o'.
ninja: build stopped: subcommand failed.
```

Error: Creation method of abstract class cannot be public.

Abstract class cannot be public, seems completely foreign to me, probably has an abstract meeting. I have two C compilers clang and gcc, and two make programs the standard make for cland and gmake for gcc.

**Comment by steveno** [comment]
That's a valac error. That's not from GCC or clang. 

What version of vala are you using?

**Comment by anoduck** [comment]
Vala 0.46.5

**Comment by steveno** [comment]
This was fixed in ed8074688e2aa4a8018b98c2b32f08cff3688d09. I just forgot to close this issue. Sorry.

**Comment by anoduck** [comment]
It worked like butter I tell you, absolute unadulterated butter.
